src/Database/FieldTypeConverter.php

```diff
@@ -67,52 +54,15 @@ class FieldTypeConverter
      */
     public function __construct(TypeMap $typeMap, DriverInterface $driver)
     {
-        $this->_driver = $driver;
-        $map = $typeMap->toArray();
-        $types = TypeFactory::buildAll();
-
-        $simpleMap = $batchingMap = [];
-        $simpleResult = $batchingResult = [];
-
-        foreach ($types as $k => $type) {
-            if ($type instanceof OptionalConvertInterface && !$type->requiresToPhpCast()) {
-                continue;
-            }
-
-            if ($type instanceof BatchCastingInterface) {
-                $batchingMap[$k] = $type;
-                continue;
-            }
-
-            $simpleMap[$k] = $type;
-        }
-
-        foreach ($map as $field => $type) {
-            if (isset($simpleMap[$type])) {
-                $simpleResult[$field] = $simpleMap[$type];
-                continue;
-            }
-            if (isset($batchingMap[$type])) {
-                $batchingResult[$type][] = $field;
-            }
-        }
-
-        // Using batching when there is only a couple for the type is actually slower,
-        // so, let's check for that case here.
-        foreach ($batchingResult as $type => $fields) {
-            if (count($fields) > 2) {
-                continue;
-            }
+        $this->driver = $driver;
 
-            foreach ($fields as $f) {
-                $simpleResult[$f] = $batchingMap[$type];
+        $types = TypeFactory::buildAll();
```

Comment:
Would be nice to remove the need to build all types whenever you request a type.

Reply:
At least `buildAll()` is a one-time build internally :shrug: